docs/design/storage-auto-scale.md

```diff
@@ -0,0 +1,83 @@
+# Storage Automatic Scaler
+
+## Summary
+
+This feature would help users to get rid of maintenance task of keeping the system with enough available free space.
+In virtualized environments, we can scale the storage devices Osds so this feature helps in automatic scaling. 
+
+## Proposal
+
+Implement auto storage scale CR per device class of a specific Storagecluster.
+
+## High-Level Design
+
+- Create a StorageAutoScale CR per Storagecluster device class
+    ```
+    apiVersion: ocs.openshift.io/v1
+    kind: StorageAutoScaling
+    metadata:
+        name: <storageclusterName>-<deviceClassName>
+        namespace: openshift-storage
+    spec:
+        storageCluster: 
+            name: <storageclusterName>
+        deviceClassName: ssd
+        clusterCapacityLimit: <capacity>
+        maxOsdSize: 8Ti
+        timeoutSeconds: 1800   
+    status:
+        phase: (NotStarted | InProgress | Failed | Succeeded)
+        LastRunTimestamp: (only on Failed | Succeeded)
+        currentOsdSize: 2TiB
+        expectedOsdSize: 4TiB
+        currentOsdCount: 3​
+        expectedOsdCount: 3
+    ```
+
+- Watch prometheus periodically every 10 mins and depending on the Osd fill percentage, reconcile the StorageAutoScaling CR.
+  
+- If the Osd size is less than maxOsdSize(default:8Tib), do vertical scaling by doubling the each osd sizes for that device class.
+
+- If the Osd size is equal to maxOsdSize(default:8Tib), do a horizontal scaling, by adding 1 osd of 8Tib on each storage node, by simply increasing the (count+1) in the device set associated with a device class.   
+
+## Low-Level Design 
+
+1) Create a controller that requeueAfter(https://pkg.go.dev/sigs.k8s.io/controller-runtime/pkg/reconcile#Result) every 10 mins.
+   
```

Comment:
1a) If an expansion is in progress, check the progress and then requeue each 1 minute until the expansion is completed successfully.